code-challenges.test.js

```diff
@@ -21,8 +24,40 @@ const secretCodeWord2 = "Gobbledygook"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+// RED
+
+describe("codeMaker", () => {
+    it("takes in a string and returns that string codified with numbers and letters", () => {
+        expect(codeMaker(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(codeMaker(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(codeMaker(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
 // b) Create the function that makes the test pass.
 
+// GREEN
+// create a function takes in a string, and codifies that string following specific parameters. a = 4, e = 3, i = 1, o = 0.
+// input: a string of letters
+// output: the same string with a, e, i, and o changed to 4, 3, 1, and 0 respectively
+// use .replaceAll() to specify which letters are changing to what and to change each instance of that letter, chaining as many times as needed to achieve the result
+
+const codeMaker = (string) => {
+    let code = string.replaceAll('a', '4').replaceAll('e', '3').replaceAll('i', '1').replaceAll('o', '0').replaceAll('A', '4').replaceAll('E', '3').replaceAll('I', '1').replaceAll('O', '0')
+    return code
+}
+console.log(codeMaker(secretCodeWord1))
+// output: L4ck4d41s1c4l
+console.log(codeMaker(secretCodeWord2))
+// output: G0bbl3dyg00k
+console.log(codeMaker(secretCodeWord3))
+// output: 3cc3ntr1c
+
+// notes: 
+// While the above code works technically, it is also super clunky and repetitive. In a potential refactor, I would love to clean this up. However, after researching, most opinions I have seen share the sentiment that it is fine to chain code so long as it is working and consistent. I will do further research to hopefully find a way to replace values more efficiently!
```

Comment:
It is this self awareness that shines confidence through you. Amazingly put, Moss!